File: 0118-pascals-triangle/0118-pascals-triangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
  if (numRows === 1) return [[1]];
  const ans = [[1], [1, 1]];
  while (numRows > 2) {
    const next = [];
    const prev = ans.at(-1);
    next.push(1);
    for (let i = 1; i < prev.length; i++) {
      next.push(prev[i - 1] + prev[i]);
    }
    next.push(1);
    ans.push(next);
    numRows--;
  }
  return ans;
};

File: 0118-pascals-triangle/0118-pascals-triangle.rb
# @param {Integer} num_rows
# @return {Integer[][]}
def generate(num_rows)
  return [[1]] if num_rows == 1
  ans = [[1], [1, 1]]
  while num_rows > 2
    prev = ans[-1]
    next_row = [1]
    (1...prev.length).each { |i| next_row << prev[i-1]+prev[i] }
    next_row << 1
    ans << next_row
    num_rows -= 1
  end
  ans
end